pkg/apis/kubeone/types.go

```diff
@@ -167,6 +167,10 @@ type HostConfig struct {
 	// PublicAddress is externally accessible IP address from public internet.
 	PublicAddress string `json:"publicAddress"`
 
+	// IPv6AddressList is IPv6 addresses of the node, only the first one will be announced to the k8s control plane.
+	// It is a list because you can request lots of IPv6 addresses (for example in case you want to assign one address per service).
+	IPv6AddressList []string `json:"IPv6AddressList"`
```

Comment:
- Why do we need this to be a list? As far as I see, we're currently consuming only the first IP address. Are there use cases where we might need other IP addresses?
- If we keep this as a list, I'd call this `IPv6Addresses` (plural, without `List` suffix). I'd also use `ipv6Addresses` as a JSON tag (lowercase `ipv6`)
- What's the type of those IPv6 addresses? Are they public or private? Do we have that distinction as we do for IPv4?

Reply:
> Why do we need this to be a list? As far as I see, we're currently consuming only the first IP address. Are there use cases where we might need other IP addresses?

Idea is that you would request whole bunch addresses and assign one to each pod/service etc.
 
This is what AWS api also does and I am closely following it because they have the best IPv6 support so far. 

>  If we keep this as a list, I'd call this IPv6Addresses (plural, without List suffix). I'd also use ipv6Addresses as a JSON tag (lowercase ipv6)

I agree. 

> What's the type of those IPv6 addresses? Are they public or private? Do we have that distinction as we do for IPv4?

No. IPv6 doesn't have public vs private distinction.

Reply:
Okay, sounds good to me. Let's just change the field name and the JSON tag.